File: lua/bob/bob.lua
bob = {}

function bob.hey(phrase)

    if phrase == nil or phrase == '' then
        return 'Fine, be that way.'
    elseif phrase:sub(#phrase, #phrase) == '?' then
        return 'Sure'
    elseif string.upper(phrase) == phrase then
        return 'Whoa, chill out!'
    else
        return 'Whatever'
    end
end

return bob

File: go/bob/bob.go
package bob // package name must match the package name in bob_test.go
import "strings"

const testVersion = 2 // same as targetTestVersion

func Hey(what string) string {

	what = strings.TrimSpace(what)
	switch {
	case len(what) == 0:
		return "Fine. Be that way!"
	case strings.ToUpper(what) == what && hasUpper(what):
		return "Whoa, chill out!"
	case what[len(what)-1] == '?':
		return "Sure."
	default:
		return "Whatever."
	}
}

// Checks if any characters are upper.
func hasUpper(w string) bool {
	for _, r := range w {
		if r > 'A' && r < 'Z' {
			return true
		}
	}
	return false
}
